core/tests/protractor_utils/AdminPage.js

```diff
@@ -268,6 +306,139 @@ var AdminPage = function() {
+
+  this.expectFileToBeDownloaded = async function() {
+    var filename = global.downloadsPath + '/topic_similarities.csv';
+    browser.driver.wait(function() {
+      return fs.existsSync(filename);
+    }, 30000);
+  };
+
+  this.clearSearchIndex = async function() {
+    await waitFor.elementToBeClickable(searchIndexClearButton,
+      'Clear search index button not clickable');
+    await searchIndexClearButton.click();
+    await browser.switchTo().alert().accept();
+  };
+
+  this.expectSearchIndexToBeCleared = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Index successfully cleared.');
+    return true;
+  };
+
+  this.flushMigrationBotContributions = async function() {
+    await waitFor.elementToBeClickable(flushMigrationBotContributionsButton,
+      'Migration bot flush contributions button not clickable');
+    await flushMigrationBotContributionsButton.click();
+    await browser.switchTo().alert().accept();
+  };
+
+  this.expectMigrationBotContributionsToBeFlushed = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Migration bot contributions successfully flushed.',
+      'Migration bot cuntributions not flushing.');
+    return true;
+  };
+
+  this.fillExtractDataForm = async function(expID, expVer, state, ans) {
+    await waitFor.pageToFullyLoad();
+    await extractDataExplorationIdInput.sendKeys(expID);
+    await extractDataExplorationVersionInput.sendKeys(expVer);
+    await extractDataStateNameInput.sendKeys(state);
+    await extractDataNumAnswersInput.sendKeys(ans);
+    await waitFor.elementToBeClickable(extractDataFormSubmitButton,
+      'Extract data form submit button not clickable');
+    await extractDataFormSubmitButton.click();
+  };
+
+  this.expectAllDataToBeExtracted = async function() {
+    await browser.getAllWindowHandles().then((handles) => {
+      expect(handles.length).toEqual(3);
+    });
+  };
+
+  this.regenerateContributionsForTopic = async function(topicId) {
+    await regenerateContributionsTopicIdInput.sendKeys(topicId);
+    await waitFor.elementToBeClickable(regenerateContributionsSubmitButton,
+      'Regenerate conributions form submit button not clickable');
+    await regenerateContributionsSubmitButton.click();
+    await browser.switchTo().alert().accept();
+  };
+
+  this.expectRegenerationError = async function(topic) {
+    var text = 'Server error: Entity for class TopicModel with id ' +
+      topic + ' not found';
+    expect(regenerationMessage.getText()).toEqual(text);
+  };
+
+  this.expectConributionsToBeRegeneratedForTopic = async function() {
+    var text = 'No. of opportunities model created: 0';
+    expect(regenerationMessage.getText()).toEqual(text);
+  };
+
+  this.sendTestEmail = async function() {
+    await waitFor.elementToBeClickable(sendEmailButton,
+      'Send email button not clickable');
+    await sendEmailButton.click();
+  };
+
+  this.expectMailToBeSent = async function() {
```

Comment:
This function name is confusing since the sending fails

Reply:
I've changed the function name to expectEmailError()